src/unix/fs.c

```diff
@@ -217,6 +217,30 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  static int no_cloexec_support;
+  int fd;
+
+#ifdef O_CLOEXEC
+  no_cloexec_support = 1;
+#endif
+
+  fd = mkstemp((char*) req->path);
+
+  if (fd == -1) {
+    return -errno;
```

Comment:
Return -1, the erorr is taken from errno. These functions must return the value which is expected to be put in req->result or -1, with errno set.